src/m365/spfx/commands/project/project-upgrade/Utils.ts

```diff
@@ -29,4 +29,7 @@ export class Utils {
       (typeof project.packageJson !== 'undefined' &&
       typeof project.packageJson.dependencies['knockout'] !== 'undefined');
   }
+  public static handleUTF8BOMEncoding(s: string): any {
```

Comment:
Let's rename it to `parseJsonWithBom` to make it clearer on the outside what the method actually does.